frontend/src/App.tsx

```diff
@@ -581,7 +580,10 @@ export class App extends PureComponent<Props, State> {
 
   createThemeHash = (themeInput: CustomThemeConfig): string | null => {
     if (!themeInput) {
-      return null
+      // Differentiate between the case where this.state.themeHash has yet to
+      // be computed (so this.state.themeHash === null)  and the case where we
+      // received no custom theme from the server.
```

Comment:
This comment and function is confusing to me. I might just be lacking context, so take this with a grain of salt, but here's how (I think) this could be made clearer:

- Change the function's return value to `: string`. That makes it clear that a non-null, non-undefined string must be returned. (I think this is already the case?)
- Change `this.state.themeHash`'s type to `themeHash?: string`. An undefined themeHash has a clear meaning: it's simply not defined yet. `this.state.themeHash` will never be null.
- Change this comment to say something like: "If themeInput is null, then the app has no custom theme. We use a hardcoded string literal for the hash in this case."
- Nit: "no_theme_input" isn't particularly descriptive. What about "undefined_theme_hash" as a string, instead?

Reply:
Thanks! I agree that changing the type of `this.state.themeHash` to be optional (and making changes for your other suggestions) does make things more clear here.